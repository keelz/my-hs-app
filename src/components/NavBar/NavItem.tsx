import * as React from 'react';
import { IComponentProps } from '../../common/models/App';
import { composeClassname } from '../../common/utils';

export interface INavItemProps extends IComponentProps {
  render?: (children: React.ReactNode) => JSX.Element;
}

const NavItem: React.SFC<INavItemProps> = props =>
  <li className={composeClassname('Nav-item')(props.className)}>
    { props.render
        ? props.render(props.children)
        : props.children
    }
  </li>;

export default NavItem;
