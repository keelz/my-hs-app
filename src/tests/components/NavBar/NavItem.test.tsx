import * as React from 'react';
import * as enzyme from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import NavItem from '../../../components/NavBar/NavItem';

describe('NavBar NavItem', () => {
  describe('snapshots', () => {
    it('renders without crashing', () => {
      const tree = enzyme.shallow(<NavItem />);
      expect(shallowToJson(tree)).toMatchSnapshot();
    });

    it('renders with children correctly', () => {
      const Component = () =>
        <NavItem><div>test</div></NavItem>;
      const tree = enzyme.shallow(<Component />);
      expect(shallowToJson(tree)).toMatchSnapshot();
    });
  });

  describe('integration', () => {
    it('renders with a render prop correctly', () => {
      const render = jest.fn();
      enzyme.shallow(<NavItem render={render} />);
      expect(render).toHaveBeenCalled();
    });
  });
});
