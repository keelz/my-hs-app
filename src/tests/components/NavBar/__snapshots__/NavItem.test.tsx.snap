// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar NavItem snapshots renders with children correctly 1`] = `
<NavItem>
  <div>
    test
  </div>
</NavItem>
`;

exports[`NavBar NavItem snapshots renders without crashing 1`] = `
<li
  className="Nav-item"
/>
`;
